lib/connection.ts

```diff
@@ -540,24 +628,27 @@ export class Connection extends Entity {
    *
    * @return Promise<AwaitableSender>.
    */
-  async createAwaitableSender(options?: AwaitableSenderOptionsWithSession): Promise<AwaitableSender> {
+  async createAwaitableSender(options?: AwaitableSenderOptionsWithSession & { abortSignal?: AbortSignalLike; }): Promise<AwaitableSender> {
     if (options && options.session && options.session.createAwaitableSender) {
       return options.session.createAwaitableSender(options);
     }
-    const session = await this.createSession();
+    const session = await this.createSession(options?.abortSignal);
     return session.createAwaitableSender(options);
   }
 
   /**
    * Creates an amqp receiver link. It either uses the provided session or creates a new one.
+   * - **Resolves** the promise with the Sender object when rhea emits the "receiver_open" event.
+   * - **Rejects** the promise with an AmqpError when rhea emits the "receiver_close" event while
+   * trying to create an amqp session or with an AbortError if the operation was cancelled.
    * @param {ReceiverOptionsWithSession} options Optional parameters to create a receiver link.
    * @return {Promise<Receiver>} Promise<Receiver>.
    */
-  async createReceiver(options?: ReceiverOptionsWithSession): Promise<Receiver> {
+  async createReceiver(options?: ReceiverOptionsWithSession & { abortSignal?: AbortSignalLike; }): Promise<Receiver> {
```

Comment:
Why isn't `abortSignal` just in the `ReceiverOptionsWithSession`?

Reply:
Answered in https://github.com/amqp/rhea-promise/pull/66/files#r464826879